src/From.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ArchTech\Enums;
+
+use ValueError;
+
+trait From
+{
+    /**
+     * Gets the Enum by name, if it exists, for "Pure" enums.
+     *
+     * This will not override the `from()` method on BackedEnums
+     *
+     * @throws ValueError
+     */
+    public static function from(string $case): static
```

Comment:
How exactly does it work that the method can be present in the `trait`, but doesn't override it on the `Enum`?

I'd expect this to cause a fatal error, but perhaps it's trait-specific behavior? Meaning, attempting to put `from()` into the `BackedEnum` itself would cause issues, but putting it in a trait just makes the `BackedEnum` ignore it and not "pull it in".

Reply:
Yep, it's just default trait behaviour - methods from a trait can't override methods with the [same name](https://3v4l.org/OFjvN) that are already defined in a class, even if it signatures don't match.

There are issues if you define _variables_ in a trait then attempt to [re-define them](https://3v4l.org/OHp0V) with a different default value in the class, but functions from a trait will only be included if not already defined.

Reply:
Awesome, that makes this feature super simple then (i.e. no need for two separate traits). Great implementation!